context/app/static/js/components/workspaces/SelectableTemplateGrid/SelectableTemplateGrid.tsx

```diff
@@ -51,19 +54,37 @@ function SelectableTemplateGrid<FormType extends FormWithTemplates>({
     [field, setSelectedTemplates],
   );
 
+  const selectOrUnselectTemplate = useCallback(
+    (templateKey: string, action: 'select' | 'unselect') => {
+      const templatesCopy = new Set(selectedTemplates);
+      if (action === 'select') {
+        templatesCopy.add(templateKey);
+      } else {
+        templatesCopy.delete(templateKey);
+      }
+      updateTemplates([...templatesCopy]);
+    },
+    [selectedTemplates, updateTemplates],
+  );
+
   const selectItem = useCallback(
     (e: ChangeEvent<HTMLInputElement>) => {
-      const templatesCopy = selectedTemplates;
       if (e.target.checked) {
-        templatesCopy.add(e.target.value);
+        selectOrUnselectTemplate(e.target.value, 'select');
       } else {
-        templatesCopy.delete(e.target.value);
+        selectOrUnselectTemplate(e.target.value, 'unselect');
       }
-      updateTemplates([...templatesCopy]);
     },
-    [updateTemplates, selectedTemplates],
+    [selectOrUnselectTemplate],
   );
 
+  useEffect(() => {
+    if (defaultSelectedTemplateKey) {
+      selectOrUnselectTemplate(defaultSelectedTemplateKey, 'select');
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
This is the `useEffect` issue mentioned above. I want this hook to be triggered on component mount only, which is why the empty dependency array is necessary - however, as this violates eslint rules, I'm assuming there is likely a better way to accomplish this. If the included dependencies are added to the array, the component will re-render infinitely.